src/middlewared/middlewared/plugins/iscsi.py

```diff
@@ -1583,25 +1587,20 @@ async def delete(self, attachments):
                 lun_ids.append(te['lunid'])
 
             await self.middleware.call('datastore.delete', 'services.iscsitargetextent', attachment['id'])
+            await self.remove_alert(attachment)
 
         await self._service_change('iscsitarget', 'reload')
 
         # For SCALE, reload action will remove existing LUN(s)
         if osc.IS_FREEBSD:
             await asyncio.sleep(5)
 
-    async def toggle(self, attachments, enabled):
-        lun_ids = []
-        for attachment in attachments:
-            for te in await self.middleware.call('iscsi.targetextent.query', [['extent', '=', attachment['id']]]):
-                lun_ids.append(te['lunid'])
-
-            await self.middleware.call('datastore.update', 'services.iscsitargetextent', attachment['id'],
-                                       {'iscsi_target_extent_enabled': enabled})
-
+    async def restart_reload_services(self, attachments):
         await self._service_change('iscsitarget', 'reload')
 
-        if osc.IS_FREEBSD and not enabled:
+    async def stop(self, attachments):
+        await self.restart_reload_services(attachments)
+        if osc.IS_FREEBSD:
```

Comment:
I think the sleep should be for restart_reload, not stop

Reply:
As per my understanding, the sleep is only required if we are removing luns/extents and is not required in the case where we add them ? That's why we have this specific to stop for now. If you think this is not the case, please let me know and we can have it in restart_reload